src/datadog/string_util.cpp

```diff
@@ -79,12 +88,15 @@ std::string join_tags(
 }
 
 bool starts_with(StringView subject, StringView prefix) {
-  if (prefix.size() > subject.size()) {
-    return false;
+  auto s = subject.data();
+  auto p = prefix.data();
+  const auto prefix_end = p + prefix.size();
+  while (*s == *p) {
+    ++s;
+    ++p;
   }
 
-  return std::mismatch(subject.begin(), subject.end(), prefix.begin()).second ==
-         prefix.end();
+  return p == prefix_end;
 }
```

Comment:
[question] why?

Reply:
There's nothing stopping `s` and `p` from going over the end of the `string_view`, so it would be undefined behavior, but even if nothing breaks the condition `p == prefix_end` will be false. Did a small example to reproduce it

<img width="684" alt="image" src="https://github.com/DataDog/dd-trace-cpp/assets/134320516/39e5b148-15d4-41b7-8f9f-58c98a713ecb">


To avoid that, the function could be written like this for example

```cpp
bool starts_with(StringView subject, StringView prefix) {
  auto s = subject.data();
  auto p = prefix.data();
  const auto prefix_end = p + prefix.size();
  while (p < prefix_end) {
    if (*s++ != *p++) {
        return false;
    }
  }

  return true;
}
```

Reply:
> There's nothing stopping s and p from going over the end of the string_view, so it would be undefined behavior
Indeed, thanks to Github diff awful diff I thought `if (prefix.size() > subject.size()) { ... }`  was part of the implementation, which reduce the area of the UB. However, as you mentioned, if `prefix.size() == subject.size()` then it's undefined behaviour and plain wrong. Thank you.